apps/Arm/vm_virtio_net/components/PingClient/ping_client.c

```diff
@@ -90,9 +90,11 @@ void print_ip_packet(void *ip_buf, size_t ip_length)
 
 int create_arp_req_reply(char *recv_data, size_t recv_data_size)
 {
-    unsigned char reply_buffer[ETHERMTU];
+    char reply_buffer[ETHERMTU];
 
-    struct ethhdr *rcv_req = (struct ethhdr *) recv_data;
+    //---------------------------------
+    //| ethhdr | ether_arp            |
+    //---------------------------------
```

Comment:
Documentation improvement are always welcome, but please make this a separate commit as it's not related to fixing the type errors

Reply:
Let's give it a bit more context:
```suggestion
    // recv_data holds a packet of the form:
    // +--------+-----------+
    // | ethhdr | ether_arp |
    // +--------+-----------+    
```